### PR TITLE
Validate the values of $merge_behavior

### PR DESCRIPTION
**Pull request by tampakrap**
This introduces dependency to puppetlabs/stdlib module

### CODE REVIEW

**Comment by elyscape** [comment]
The dependency on stdlib isn't necessary. Instead of using `validate_re`, you should be able to do something like:
```puppet
unless $merge_behavior in ['native', 'deep', 'deeper'] {
  fail("Invalid merge behavior: ${merge_behavior}")
}
```

**Comment by tampakrap** [comment]
Thanks for the review, I rebased my PR

File: manifests/init.pp [hunk 1]
```diff
@@ -134,6 +134,10 @@
       ensure => directory,
     }
   }
+  if $merge_behavior {
+    unless $merge_behavior in ['native', 'deep', 'deeper'] {
+      fail("${merge_behavior} merge behavior is invalid. Valid values are: native, deep, deeper")
```
Comment (igalic) [inline manifests/init.pp]: this is now missing a `}`

Comment (tampakrap) [inline manifests/init.pp]: thanks, fixed

File: manifests/init.pp [hunk 2]
```diff
@@ -134,6 +134,11 @@
...
+    }
```
Comment (igalic) [inline manifests/init.pp]: now wondering if puppet optimizes this down to:
```puppet
if $merge_behavior and !($merge_behavior in ['native', 'deep', 'deeper']) {
```
:P

Comment (igalic) [inline manifests/init.pp]: logic is hard…

Comment (elyscape) [inline manifests/init.pp]: Hah, good point.  That might be a better way to specify that check.

**Pull request merged**